tests/manual-test-cases/Group5-Functional-Tests/5-3-Enhanced-Linked-Mode.robot

```diff
@@ -41,18 +51,39 @@ Test
     \   ${ip}=  Run Keyword If  ${status}  Fetch From Right  ${line}  ${SPACE}
     \   Run Keyword If  ${status}  Set Test Variable  ${esx3-ip}  ${ip}
 
-    ${esx1}  ${esx4-ip}  ${esx2}  ${esx5-ip}  ${esx3}  ${esx6-ip}=  Deploy Multiple Nimbus ESXi Servers in Parallel  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
+    &{esxes}=  Create Dictionary
+    ${num_of_esxes}=  Evaluate  3
+    :FOR  ${i}  IN RANGE  3
+    # Deploy some ESXi instances
+    \    &{new_esxes}=  Deploy Multiple Nimbus ESXi Servers in Parallel  ${num_of_esxes}  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
+    \    ${esxes}=  Combine Dictionaries  ${esxes}  ${new_esxes}
+
+    # Investigate to see how many were actually deployed
+    \    ${len}=  Get Length  ${esxes}
+    \    ${num_of_esxes}=  Evaluate  3 - ${len}
+
+    # Exit if we've got enough & continue loop if we don't
+    \    Exit For Loop If  ${len} >= 3
+    \    Log To Console  Only got ${len} ESXi instance(s); Trying again
+
+    @{esx-names}=  Get Dictionary Keys  ${esxes}
```

Comment:
This bit could be refactored away but we'd have to refactor the rest of the test following these assignments and I did not think that was worth the time.